pkg/plugins/policies/meshtrace/api/v1alpha1/meshtrace.proto

```diff
@@ -0,0 +1,133 @@
+syntax = "proto3";
+
+package kuma.plugins.policies.meshtrace.v1alpha1;
+
+import "mesh/options.proto";
+option go_package = "github.com/kumahq/kuma/pkg/plugins/policies/meshtrace/api/v1alpha1";
+
+import "common/v1alpha1/targetref.proto";
+import "google/protobuf/wrappers.proto";
+import "kuma-doc/config.proto";
+
+option (doc.config) = {
+  type : Policy,
+  name : "MeshTrace",
+  file_name : "meshtrace"
+};
+
+// MeshTrace
+message MeshTrace {
+  option (kuma.mesh.policy) = {
+    // Toggle this to have the policy registered or not in Kuma
+    skip_registration : false,
+  };
+
+  // TargetRef is a reference to the resource the policy takes an effect on.
+  // The resource could be either a real store object or virtual resource
+  // defined inplace.
+  kuma.common.v1alpha1.TargetRef targetRef = 1;
+
+  // Datadog tracing backend configuration.
+  message DatadogBackend {
+    // Address of datadog collector.
+    string address = 1 [ (doc.required) = true ];
+
+    // Port of datadog collector
+    uint32 port = 2 [ (doc.required) = true ];
+  }
+
+  // Zipkin tracing backend configuration.
+  message ZipkinBackend {
+    // Address of Zipkin collector.
+    string url = 1 [ (doc.required) = true ];
+
+    // Generate 128bit traces. Default: false
+    bool traceId128bit = 2;
+
+    // Version of the API. values: httpJson, httpProto. Default:
+    // httpJson see
+    // https://github.com/envoyproxy/envoy/blob/v1.22.0/api/envoy/config/trace/v3/zipkin.proto#L66
+    string apiVersion = 3 [ (doc.required) = true ];
+
+    // Determines whether client and server spans will share the same span
+    // context. Default: true.
+    // https://github.com/envoyproxy/envoy/blob/v1.22.0/api/envoy/config/trace/v3/zipkin.proto#L63
+    google.protobuf.BoolValue sharedSpanContext = 4;
+  }
+
+  // Only one of zipkin or datadog can be used.
+  message Backend {
+    // Zipkin backend configuration.
+    ZipkinBackend zipkin = 1;
+    // Datadog backend configuration.
+    DatadogBackend datadog = 2;
+  }
+
+  // Sampling configuration.
+  message Sampling {
+    // Target percentage of requests will be traced
+    // after all other sampling checks have been applied (client, force tracing,
+    // random sampling). This field functions as an upper limit on the total
+    // configured sampling rate. For instance, setting client_sampling to 100%
+    // but overall_sampling to 1% will result in only 1% of client requests with
+    // the appropriate headers to be force traced. Default: 100% Mirror of
+    // overall_sampling in Envoy
+    // https://github.com/envoyproxy/envoy/blob/v1.22.0/api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto#L142-L150
+    uint32 overall = 1;
```

Comment:
Percent type in Envoy is `double` https://www.envoyproxy.io/docs/envoy/latest/api-v3/type/v3/percent.proto#envoy-v3-api-msg-type-v3-percent. Do we want to have `double` as well here?

Reply:
This is why we decided to use `uint32` - https://github.com/kumahq/kuma/pull/5069#discussion_r982438454